scripts/generate

```diff
@@ -391,23 +414,26 @@ const generate =
 def ('generate')
     ({})
     ([$.String, Html.Type])
-    (pipe ([replace (regex ('gm') ('^(#{1,6}) <span id="([^"]*)">❑ (.*)</span>$'))
-                    (($0, $1, $2, $3) => `<h${$1.length} id="${$2}">${$3}</h${$1.length}>`),
-            replace (regex ('gm') ('^#### <a name="(.*)" href="(.*)">`(.*)`</a>$'))
-                    (($0, $1, $2, $3) => Html.unwrap (headingToMarkup ($1) ($2) ($3))),
-            replace (regex ('gm') ('^```javascript\\n(> [^]*?)^```\\n'))
-                    (($0, $1) => Html.unwrap (doctestsToMarkup ($1))),
+    (pipe ([replaceBy (contramap (justs) (([$1, $2, $3]) => `<h${$1.length} id="${$2}">${$3}</h${$1.length}>`))
+                      (regex ('gm') ('^(#{1,6}) <span id="([^"]*)">❑ (.*)</span>$')),
+            replaceBy (contramap (justs) (([$1, $2, $3]) => Html.unwrap (headingToMarkup ($1) ($2) ($3))))
+                      (regex ('gm') ('^#### <a name="(.*)" href="(.*)">`(.*)`</a>$')),
+            replaceBy (contramap (justs) (([$1]) => Html.unwrap (doctestsToMarkup ($1))))
+                      (regex ('gm') ('^```javascript\\n(> [^]*?)^```\\n')),
             s => marked (s, {}),
-            replace (regex ('g') (ZERO_WIDTH_SPACE)) (''),
-            replace (/&#39;/g) ("'"),
-            replace (/&gt;/g) ('>'),
-            replace (regex ('gm') ('(?!^)(?=</code></pre>)')) ('\n'),
-            replace (regex ('g') ('(<pre><code class="language-javascript">)([^]*?)(?=</code></pre>)'))
-                    (($0, $1, $2) => $1 + Html.unwrap (highlight (Html ($2)))),
-            replace (regex ('gm') ('</code></pre>(?!$)')) ($0 => $0 + '\n'),
-            replace (regex ('g') ('<(h[2-6]) id="([^"]*)">(.*)</\\1>\n*'))
-                    ('<a class="pilcrow $1" href="#$2">' + PILCROW_SIGN + '</a>\n' +
-                     '<$1 id="$2">$3</$1>\n'),
+            replaceWith ('') (regex ('g') (ZERO_WIDTH_SPACE)),
+            replaceWith ("'") (/&#39;/g),
+            replaceWith ('>') (/&gt;/g),
+            replaceWith ('\n') (regex ('gm') ('(?!^)(?=</code></pre>)')),
+            replaceBy (contramap (justs) (([$1, $2]) => $1 + Html.unwrap (highlight (Html ($2)))))
+                      (regex ('g') ('(<pre><code class="language-javascript">)([^]*?)(?=</code></pre>)')),
+            replaceBy (contramap (justs) (([$1]) => $1 + '\n'))
+                      (regex ('gm') ('(</code></pre>)(?!$)')),
+            replaceBy (contramap (justs)
+                                 (([tagName, id, markup]) =>
+                                    `<a class="pilcrow ${tagName}" href="#${id}">${PILCROW_SIGN}</a>\n` +
+                                    `<${tagName} id="${id}">${markup}</${tagName}>\n`))
+                      (regex ('g') ('<(h[2-6]) id="([^"]*)">(.*)</\\1>\n*')),
```

Comment:
This demonstrates an advantage of `replaceBy` over [special replacement patterns][1]: one can give captured values descriptive names.


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter